cas-server-webapp/src/main/java/org/jasig/cas/web/support/AbstractThrottledSubmissionHandlerInterceptorAdapter.java

```diff
@@ -0,0 +1,177 @@
+/*
```

Comment:
This class overwrites class [AbstractThrottledSubmissionHandlerInterceptorAdapter](https://github.com/apereo/cas/blob/4.1.x/cas-server-webapp-support/src/main/java/org/jasig/cas/web/support/AbstractThrottledSubmissionHandlerInterceptorAdapter.java), and is extended by class [AbstractInMemoryThrottledSubmissionHandlerInterceptorAdapter](https://github.com/apereo/cas/blob/4.1.x/cas-server-webapp-support/src/main/java/org/jasig/cas/web/support/AbstractInMemoryThrottledSubmissionHandlerInterceptorAdapter.java), which is further extended by the concrete class we use for login throttle: [InMemoryThrottledSubmissionByIpAddressAndUsernameHandlerInterceptorAdapter](https://github.com/apereo/cas/blob/4.1.x/cas-server-webapp-support/src/main/java/org/jasig/cas/web/support/InMemoryThrottledSubmissionByIpAddressAndUsernameHandlerInterceptorAdapter.java).